sympy/integrals/transforms.py

```diff
@@ -1008,9 +1008,36 @@ def repl(ex, *args):
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, limit,
+                       Add, DiracDelta)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    zero_minus = symbols('zero_minus', real=True)
+    # Expressions with DiracDelta need to be computed with a directional
+    # limit, but this will fail on some sums, so we first use the
+    # linearity of the Laplace transform to split the integral into a sum
+    # of integrals.
+    if f.has(DiracDelta):
+        if f.func != Add:
```

Comment:
Why this test?

Reply:
It saves the effort of expanding when it is already expanded. Do you think it could be omitted, and expand executed without checking first?

Reply:
If `e` is any expression, then `e + 1` would not be expanded here.

Reply:
> If `e` is any expression, then `e + 1` would not be expanded here.

You are right, I will change it.

Reply:
I have run my local tests; since I have restricted what `expand` shall do, I do not see a problem or speed penalty from omitting 'if f.func != Add:'